spec/controllers/concerns/idv/threat_metrix_concern_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Idv::ThreatMetrixConcern, type: :controller do
+  controller ApplicationController do
+    include Idv::ThreatMetrixConcern
+
+    before_action :override_csp_for_threat_metrix
+
+    def index; end
+  end
+
+  describe '#override_csp_for_threat_metrix' do
+    context 'non-production environment' do
+      before do
+        allow(Rails.env).to receive(:production?).and_return(false)
+      end
+      it 'does not modify CSP headers for SSN step' do
+        assert_csp_is_not_modified 'ssn'
+      end
+      it 'does not modify CSP headers for any other step' do
+        assert_csp_is_not_modified 'some_other_step'
+      end
+    end
+
+    context 'production enviroment' do
+      before do
+        allow(Rails.env).to receive(:production?).and_return(true)
+      end
+
+      context 'ff is set' do
+        before do
+          allow(IdentityConfig.store).to receive(:proofing_device_profiling_collecting_enabled).
+            and_return(true)
+        end
+        it 'modifies CSP headers for SSN step' do
+          assert_csp_is_modified 'ssn'
+        end
+
+        it 'does not modify CSP headers for any other step' do
+          assert_csp_is_not_modified 'some_other_step'
+        end
+      end
+
+      context 'ff is not set' do
+        before do
+          allow(IdentityConfig.store).to receive(:proofing_device_profiling_collecting_enabled).
+            and_return(false)
+        end
+
+        it 'does not modify CSP headers for SSN step' do
+          assert_csp_is_not_modified 'ssn'
+        end
+        it 'does not modify CSP headers for any other step' do
+          assert_csp_is_not_modified 'some_other_step'
+        end
+      end
+    end
+  end
+
+  private
+
+  def assert_csp_is_modified(step)
+    get :index, params: { step: step }
+
+    csp = response.request.content_security_policy
+
+    expect(csp.directives['script-src']).to include('h.online-metrix.net')
+    expect(csp.directives['script-src']).to include("'unsafe-eval'")
+
+    expect(csp.directives['style-src']).to include("'unsafe-inline'")
+
+    expect(csp.directives['child-src']).to include('h.online-metrix.net')
+
+    expect(csp.directives['connect-src']).to include('h.online-metrix.net')
+
+    expect(csp.directives['img-src']).to include('*.online-metrix.net')
```

Comment:
totally optional but sometimes makes it easier when running specs with lots of assertions, this will group all assertions but still continue in the spec

```suggestion
    aggregate_failures do
      expect(csp.directives['script-src']).to include('h.online-metrix.net')
      expect(csp.directives['script-src']).to include("'unsafe-eval'")

      expect(csp.directives['style-src']).to include("'unsafe-inline'")

      expect(csp.directives['child-src']).to include('h.online-metrix.net')

      expect(csp.directives['connect-src']).to include('h.online-metrix.net')

      expect(csp.directives['img-src']).to include('*.online-metrix.net')
    end
```